test/extended/networking/egress_firewall.go

```diff
@@ -114,6 +117,19 @@ func doEgressFwTest(f *e2e.Framework, mgmtFw *e2e.Framework, oc *exutil.CLI, man
 	err := oc.AsAdmin().Run("create").Args("-f", egFwYaml).Execute()
 	o.Expect(err).NotTo(o.HaveOccurred(), "created egress-firewall object")
 
+	g.By("waiting for egressfirewall rules to be applied successfully")
+	// Fetch the name of the egressfirewall object. As there can be only
+	// one egressfirewall object per namespace, we can use the first one.
+	egfwName, err := oc.AsAdmin().Run("get").Args("egressfirewall", "-o", "jsonpath={.items[0].metadata.name}").Output()
+	o.Expect(err).NotTo(o.HaveOccurred(), "failed to get egressfirewall object")
+
+	err = wait.PollUntilContextTimeout(context.TODO(), 100*time.Millisecond, 1*time.Second, true, func(ctx context.Context) (bool, error) {
+		out, err := oc.AsAdmin().Run("get").Args("egressfirewall", egfwName, "-o", "jsonpath={.status.status}").Output()
+		o.Expect(err).NotTo(o.HaveOccurred(), "failed to get egressfirewall object")
```

Comment:
I think we want to keep retrying on failure until success or timeout. So we'd better just return false here.

Reply:
Thanks for pointing that out. I have made the changes.